docs/api-documentation.md

```diff
@@ -0,0 +1,370 @@
+}
+```
+
+### Forget Password
+
+Allows the user to send forget password request to the portal and receive the reset token that is used to change the password from the API. Requires the user email.
+
+**Forgot Password Flow**
+
+Here is a step-by-step breakdown of the forgot password flow:
+
+1. **Client-Side Request:**
+
+   - The client sends a forgot password request to the server.
+
+2. **AuthenticationController Handling:**
+
+   - The `AuthenticationController` captures the forgot password request through the `POST /forgot-password` endpoint.
+
+3. **Service Delegation:**
+
+   - The `AuthenticationController` delegates the forgot password process to the `AuthenticationService`.
+
+4. **AuthenticationService Processing:**
+
+   - The `AuthenticationService` is responsible for initiating the forgot password process.
+   - It interacts with the `UserRepository` to find the user by email.
+
+5. **UserRepository Database Interaction:**
+
+   - The `UserRepository` interacts with the database (Entity Framework) to:
+     - Use `UserManager.FindByEmailAsync` to find a user by email.
+     - Generate a password reset token using `GeneratePasswordResetTokenAsync` provided by `UserManager` Component of Asp.Net Identity.
+
+6. **Email Notification:**
+
+   - The generated password reset token is sent to the user's email for security verification.
+   - `SendForgetPasswordEmailAsync` funciton of `IEmailRepository` is used for sending the email.
+   - We have integrated `osmo-x` to send email
+
+7. **Response to Client:**
+
+   - The `AuthenticationController` responds to the client with a message indicating that the password reset process has been initiated.
+
+8. **ASP.NET Identity Components Involved:**
+
+   - `UserManager<IdentityApplicationUser>`: Manages user-related operations.
+   - `IEmailRepository`: Interface for sending emails.
+
+The forgot password flow involves the client initiating a request, which is handled by the `AuthenticationController`. The controller delegates the process to the `AuthenticationService`, and the necessary steps are taken in the `UserRepository`. A password reset token is generated, sent to the user's email for verification, and a response is provided to the client. This ensures a secure and structured forgot password process using ASP.NET Identity and Entity Framework components.
+
+**Database Table**
+
+Asp.Net User Table
+![Asp.Net User Table](./images/db_table_aspnetusers.png)
+
+**Endpoint:** `http://localhost:5000/api/auth/forgot-password`
+
+**Method: `POST`**
+
+**QueryParam:**
+
+```
+email = example@gmail.com
+```
+
+**Curl**
+
+```sh
+curl -X 'POST' \
+  'http://localhost:5000/api/auth/forgot-password?email=example%40gmail.com' \
+  -H 'accept: */*' \
+  -d ''
+```
+
+**Sample response**
+
+```json
+200: Success
+```
+
+**Sample Mail**
+
+![Forget Password](./images/mail_forgetpassword.png)
+
+### Reset Password
```

Comment:
This should be "Change Password"?

Reply:
Reset Password is the next step in the forget password flow where user create a new password for himself 

Change password is sounding like the user is updating the existing password that he know